### PR TITLE
[Storage] BREAKING CHANGE: az storage account create: Change default storage account kind to StorageV2

### PR DESCRIPTION
**Pull request by Juliehzl**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---
**Description**:

The feature is required for a long time. Given there is more and more feature only available in StorageV2 Kind, we need to switch the defualt kind to StoargeV2, which will be align with Powershell, two.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -154,8 +154,10 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('location', get_location_type(self.cli_ctx), validator=get_default_location_from_resource_group)
         c.argument('account_type', help='The storage account type', arg_type=get_enum_type(t_account_type))
         c.argument('account_name', acct_name_type, options_list=['--name', '-n'], completer=None)
-        c.argument('kind', help='Indicates the type of storage account.',
-                   arg_type=get_enum_type(t_kind, default='storage'))
+        c.argument('kind', help='Indicates the type of storage account.', min_api="2017-10-01",
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: Can you share the reason about changing the default to V2?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: Most features are only available for storageV2 storage account. In this way, StorageV2 is a general purpose storage account.
In addition, there are requirements from other team to have such default value and it is the same with PSH behavior.

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -154,8 +154,10 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+                   arg_type=get_enum_type(t_kind), default='StorageV2')
+        c.argument('kind', help='Indicates the type of storage account.', max_api="2017-06-01",
+                   arg_type=get_enum_type(t_kind), default='Storage')
```
Comment (qianwens) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: >Storage [](start = 60, length = 7)

the default value was "storage" while it is "Storage" now, is it by design?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: In cli commands, if you use get_enum_types(), the input value will be case insensitive. In this way, it is doesn't matter for "storage: or "Storage" here.

Comment (qianwens) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: so this default value is only used by our code, it is not visible to user?

---
In reply to: [376912505](https://github.com/Azure/azure-cli/pull/12112#discussion_r376912505) [](ancestors = 376912505)

**Comment by yonzhan** [comment]
add to S165.

**Pull request merged**